nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -455,7 +455,7 @@ impl DataStore {
                 let lldp_link_ids: Vec<Uuid> = result
                     .links
                     .iter()
-                    .map(|link| link.lldp_link_config_id)
+                    .filter_map(|link| link.lldp_link_config_id)
```

Comment:
This file deserves extra scrutiny - this function (`switch_port_settings_get`) and the call below (`do_switch_port_settings_delete`) can now see an empty vec of `lldp_link_ids` if this field is nullable.